### PR TITLE
ISSUE-305: [Enhancement] Ability to analyse Drupal database connections configurations

### PR DESCRIPTION
**Pull request by omkar-pednekar**
Issue: https://github.com/drutiny/drutiny/issues/305

### CODE REVIEW

File: src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Provides all database connection information.
+ */
+class DatabaseConnectionAnalysis extends AbstractAnalysis {
+
+  public function gather(Sandbox $sandbox) {
+    $db_connections = $this->target->getService('drush')
+      ->runtime(function() {
+        global $databases;
+        return $databases;
```
Comment (fiasco) [inline src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php]: I think we should copy $databases and remove credentials on that copy before returning it. That means the removed credentials never leave the Drupal runtime environment.

File: src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    $this->unset_user_pass($db_connections, ['username', 'password']);
+    $this->set('database_connections', $db_connections);
+  }
+
+  /**
+   * Helper function to unset non-required keys from connections array.
+   * @param array $connections
+   * @param array $unwanted_keys
+   */
+  public function unset_user_pass(array &$connections, array $unwanted_keys) {
```
Comment (fiasco) [inline src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php]: Unsetting credentials here means the credential data has already been transfered back to Drutiny. So while it won't be available at the templating layer, it has been aggregated and stored in the caching layer. See comment above as to why we should remove them in the runtime environment instead.

File: src/Plugin/Drupal8/Audit/Database/DatabaseConnectionAnalysis.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Provides all database connection information.
+ */
+class DatabaseConnectionAnalysis extends AbstractAnalysis {
+
+  public function gather(Sandbox $sandbox) {
+    $db_connections = $this->target->getService('drush')
+      ->runtime(function() {
+        return \Drupal\Core\Database\Database::getAllConnectionInfo();
```
Comment (fiasco) [inline src/Plugin/Drupal8/Audit/Database/DatabaseConnectionAnalysis.php]: Same as other comments in this review, lets scrub the credentials we don't want to expose before returning them.

**Review by fiasco — CHANGES_REQUESTED** [review]

File: src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php (hunk 1 above)
Comment (omkar-pednekar) [reply src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php]: Done.

File: src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php (hunk 2 above)
Comment (omkar-pednekar) [reply src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php]: Done.

File: src/Plugin/Drupal8/Audit/Database/DatabaseConnectionAnalysis.php (hunk 3 above)
Comment (omkar-pednekar) [reply src/Plugin/Drupal8/Audit/Database/DatabaseConnectionAnalysis.php]: Done.

**Comment by omkar-pednekar** [comment]
@fiasco , Could you please review it again. I have made changes as you suggested.

**Pull request merged**